packages/client/components/NewGitLabIssueInput.tsx

```diff
@@ -0,0 +1,265 @@
+import {PALETTE} from '~/styles/paletteV3'
+import getNonNullEdges from '~/utils/getNonNullEdges'
+import {NewGitLabIssueInput_viewer$key} from '~/__generated__/NewGitLabIssueInput_viewer.graphql'
+import useForm from '../hooks/useForm'
+import {PortalStatus} from '../hooks/usePortal'
+import CreateTaskMutation from '../mutations/CreateTaskMutation'
+import UpdatePokerScopeMutation from '../mutations/UpdatePokerScopeMutation'
+import {CompletedHandler} from '../types/relayMutations'
+import convertToTaskContent from '../utils/draftjs/convertToTaskContent'
+import Legitity from '../validation/Legitity'
+import {CreateTaskMutationResponse} from '../__generated__/CreateTaskMutation.graphql'
+import Checkbox from './Checkbox'
+import Icon from './Icon'
+import NewGitLabIssueMenu from './NewGitLabIssueMenu'
+import PlainButton from './PlainButton/PlainButton'
+import StyledError from './StyledError'
+
+const StyledButton = styled(PlainButton)({
+  alignItems: 'center',
+  backgroundColor: 'transparent',
+  display: 'flex',
+  height: '20px',
+  justifyContent: 'flex-start',
+  margin: 0,
+  opacity: 1,
+  ':hover, :focus': {
+    backgroundColor: 'transparent'
+  }
+})
+
+const StyledIcon = styled(Icon)({
+  color: PALETTE.SKY_500,
+  fontSize: 20,
+  padding: 0,
+  alignContent: 'center'
+})
+
+const StyledLink = styled('a')({
+  color: PALETTE.SKY_500,
+  display: 'block',
+  fontSize: 12,
+  lineHeight: '20px',
+  textDecoration: 'none',
+  '&:hover,:focus': {
+    textDecoration: 'underline'
+  }
+})
+
+const Form = styled('form')({
+  display: 'flex',
+  flexDirection: 'column',
+  width: '100%'
+})
+
+const Item = styled('div')({
+  backgroundColor: PALETTE.SLATE_100,
+  cursor: 'pointer',
+  display: 'flex',
+  paddingLeft: 16,
+  paddingTop: 8,
+  paddingBottom: 8
+})
+
+const Issue = styled('div')({
+  display: 'flex',
+  flexDirection: 'column',
+  paddingLeft: 16,
+  width: '100%'
+})
+
+const NewIssueInput = styled('input')({
+  appearance: 'none',
+  background: 'transparent',
+  border: 'none',
+  color: PALETTE.SLATE_700,
+  fontSize: 16,
+  margin: 0,
+  padding: '0px 8px 0px 0px',
+  outline: 0,
+  width: '100%'
+})
+
+const Error = styled(StyledError)({
+  fontSize: 13,
+  textAlign: 'left',
+  width: '100%'
+})
+
+interface Props {
+  isEditing: boolean
+  meetingId: string
+  setIsEditing: (isEditing: boolean) => void
+  viewerRef: NewGitLabIssueInput_viewer$key
+}
+
+const validateIssue = (issue: string) => {
+  return new Legitity(issue).trim().min(2, `C’mon, you call that an issue?`)
+}
+
+const NewGitLabIssueInput = (props: Props) => {
+  const {isEditing, meetingId, setIsEditing, viewerRef} = props
+  const viewer = useFragment(
+    graphql`
+      fragment NewGitLabIssueInput_viewer on User {
+        id
+        team(teamId: $teamId) {
+          id
+        }
+        teamMember(teamId: $teamId) {
+          id
+          integrations {
+            gitlab {
+              api {
```

Comment:
Unlike the GitHub and Jira integrations, I'm querying GitLab directly. [See here](https://github.com/ParabolInc/parabol/issues/6033#issuecomment-1069039404) for detailed explanation. TL;DR: repo integrations grabs all possible integrations rather than just the GitLab/Jira/GitHub ones that we want.